multicluster/controllers/multicluster/member/clusterset_controller.go

```diff
@@ -90,57 +93,113 @@ func NewMemberClusterSetReconciler(client client.Client,
 func (r *MemberClusterSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	clusterSet := &mcv1alpha2.ClusterSet{}
 	err := r.Get(ctx, req.NamespacedName, clusterSet)
-	r.commonAreaLock.Lock()
-	defer r.commonAreaLock.Unlock()
+	var clusterSetNotFound bool
 	if err != nil {
 		if !apierrors.IsNotFound(err) {
 			return ctrl.Result{}, err
 		}
+		clusterSetNotFound = true
+	}
+
+	cleanUpResources := func(isNotFound bool) error {
+		// Any ResourceExports belongs to this member cluster will be cleaned up by the leader cluster
+		// when its MemberClusterAnnounce is deleted.
+		if err := r.deleteMemberClusterAnnounce(ctx); err != nil {
+			// MemberClusterAnnounce could be kept in the leader cluster, if antrea-mc-controller crashes after the failure.
+			// Leader cluster will delete the stale MemberClusterAnnounce with a garbage collection mechanism in this case.
+			return fmt.Errorf("failed to delete MemberClusterAnnounce in the leader cluster: %v", err)
+		}
+		r.remoteCommonArea.Stop()
+		r.remoteCommonArea = nil
+		r.clusterSetConfig = nil
```

Comment:
I feel we should reset `r.installedLeader` too.

Reply:
Yeah, this is reset to empty.